script/core/AssetManager.js

```diff
@@ -0,0 +1,118 @@
+import SpriteAtlas from 'core/SpriteAtlas';
+import FileManager from 'core/FileManager';
+
+const SPRITE = 'SPRITE';
+
+const assets = new Map();
+let spriteIndex;
+
+export default class AssetManager {
```

Comment:
Een class met alleen static methods hoeft geen class te zijn toch? :)

Reply:
Op dit moment moet je per de spriteindex laden voordat dit object iets doet toch? Dan zou je de `url` naar de spriteindex required kunnen noemen en het een parameter van een constructor kunnen maken.
```
const assetManager = new AssetManager(url);
```
En dan is het wel weer logisch dat het een class is :)

Reply:
Ja, zeker als het maar 1 type asset behandeld (`SpriteLoader` is een top idee). Dat is sowieso wel fijner denk ik. M'n andere asset wordt waarschijnlijk muziek, en verder weet ik nog niet wat anders, dus dan is 1 loader per type prima. Ik zou uiteindelijk nog kunnen kijken naar een `Loader` base class mocht er veel shared functionality zijn, maar dat is nu nog niet nodig.

Reply:
De reden dat ik SpriteIndex trouwens zo 'static' heb opgezet, is omdat ik het wil kunnen `import`en, maar dan toegang hebben tot een state die geset is in een andere context. Als ik hem telkens moet instantiaten dan werkt dat niet lekker meer. Dit patroon gebruik ik ook in mijn `core/PubSub.js`. Als je hier een beter idee voor hebt hoor ik het graag.